shell/platform/darwin/ios/framework/Source/FlutterPlatformPlugin.mm

```diff
@@ -241,11 +241,14 @@ - (void)popSystemNavigator:(BOOL)isAnimated {
   // in the navigation hierarchy.
   // It's also possible in an Add2App scenario that the FlutterViewController was presented
   // outside the context of a UINavigationController, and still wants to be popped.
-  UIViewController* viewController = [UIApplication sharedApplication].keyWindow.rootViewController;
-  if ([viewController isKindOfClass:[UINavigationController class]]) {
-    [((UINavigationController*)viewController) popViewControllerAnimated:isAnimated];
+
+  auto engineViewController = static_cast<UIViewController*>([_engine.get() viewController]);
+  UINavigationController* navigationController = [engineViewController navigationController];
```

Comment:
Should this fall back on `UIApplication.sharedApplication.keyWindow.rootViewController` if it's `nil` to match the old behavior?

Reply:
Isn't that what it's doing on line 250?

Reply:
That code path isn't calling `popViewControllerAnimated`.

Reply:
> Should this fall back on `UIApplication.sharedApplication.keyWindow.rootViewController` if it's `nil` to match the old behavior?

Em, I think we can't match the old behavior, because if `UIApplication.sharedApplication.keyWindow.rootViewController`  is `UINavigationController` but `FlutterViewController`'s `navigationController` is nil, like `FlutterViewController` is presented, it would pop the top native view controller but not presented `FlutterViewController`

Reply:
Ah, I see, thanks for the explanation.